### PR TITLE
Reports: XML template fixes

### PR DESCRIPTION
**Pull request by psiinon**
Fixed issue where report was not generated via the API.
Made more backwards compatible as per #2952
Fixed issue where you could end up with report.xml.xml (applies to all reports).
Added more detailed tests for the report.

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Review by thc202 — APPROVED** [review]

**Comment by psiinon** [comment]
Fixed conflicts

**Comment by psiinon** [comment]
@kingthorin @ricekot 2nd review? :)

**Comment by kingthorin** [comment]
/ACK

Needs spotlessApply

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/automation/ReportJob.java [hunk 1]
```diff
@@ -72,9 +72,11 @@ public void runJob(
         // Work out the file name based on the pattern
         String fileName =
                 ExtensionReports.getNameFromPattern(
-                                reportFile, env.getDefaultContextWrapper().getUrls().get(0))
-                        + "."
-                        + template.getExtension();
+                        reportFile, env.getDefaultContextWrapper().getUrls().get(0)); // TODO.getUrlStringForContext(env.getDefaultContext()));
```
Comment (kingthorin) [inline addOns/reports/src/main/java/org/zaproxy/addon/reports/automation/ReportJob.java:75]: Was this for now or later?

Comment (psiinon) [reply addOns/reports/src/main/java/org/zaproxy/addon/reports/automation/ReportJob.java:75]: Doh - now :P

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Thanks

**Pull request merged**